gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/ModDateTimeDatasetVersionFinder.java

```diff
@@ -0,0 +1,51 @@
+package gobblin.data.management.retention.version.finder;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.TimestampedDatasetVersion;
+
+
+/**
+ * {@link DatasetVersionFinder} for datasets based on modification timestamps.
+ */
+public class ModDateTimeDatasetVersionFinder extends DatasetVersionFinder<TimestampedDatasetVersion> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ModDateTimeDatasetVersionFinder.class);
+
+  private static final Path GLOB_PATTERN = new Path(".");
```

Comment:
You can actually use `SingleVersionFinder` for this. It uses FileStatusDatasetVersion which does not require an additional call to `fs.filestatus`